client/components/util/util.service.js

```diff
@@ -47,7 +47,29 @@ function UtilService($window) {
           url.protocol === o.protocol;
       });
       return (origins.length >= 1);
-    }
+    },
+
+    formatAMPM(date) {
+      var hours = date.getHours();
+      var minutes = date.getMinutes();
+      var ampm = hours >= 12 ? 'pm' : 'am';
+      hours = hours % 12;
+      hours = hours ? hours : 12; // the hour '0' should be '12'
+      minutes = minutes < 10 ? '0'+minutes : minutes;
+      var strTime = hours + ':' + minutes + ' ' + ampm;
+      return strTime + " " + date.toDateString();
+    },
+    /**
+     * Convert Date strings to Date Objects
+     */
+    convertDates(events) {
+      for(var i=0; i< events.length; i++){
+        for(var j=0; j< events[i].info.times.length; j++){
+          events[i].info.times[j].start = Util.formatAMPM(new Date(events[i].info.times[j].start));
+          events[i].info.times[j].end = Util.formatAMPM(new Date(events[i].info.times[j].end));
+        }
```

Comment:
I'm not sure it's best to have a function called `convertDates` that manipulates an event object in place... the function must be read to understand what it does